### PR TITLE
[EOSF-623] domain redirects

### PR DESCRIPTION
**Pull request by binoculars**
### ❗️ Needs:
- [x] CenterForOpenScience/osf.io#7167
- [x] CenterForOpenScience/ember-osf#203

## Ticket

https://openscience.atlassian.net/browse/EOSF-623

## Purpose

Redirect only if domain_redirect_enabled field is true on the preprint provider

## Changes

- Updates the ember-cli-build and detects if the OSF URL in the current origin
- add-domains script now uses the API to get the domains for /etc/hosts
- remove domains from config/environment (no longer needed)
- router updated to set theme id on init from the API
- theme service updated to redirect to the domain if domainRedirectEnabled = true for the provider (only needed for local dev)
- Simplify redirects and remove get-redirect-url util

### CODE REVIEW

File: app/routes/provider.js [hunk 1]
```diff
@@ -1,6 +1,5 @@
-import Ember from 'ember';
+import Ember, { Logging } from 'ember';
```
Comment (jamescdavis) [inline app/routes/provider.js]: s/Logging/Logger/

File: app/router.js [hunk 2]
```diff
@@ -1,34 +1,29 @@
...
 
-        if (provider) {
-            this.set('theme.id', provider.id);
-            this.set('theme.isDomain', true);
+        // Set the provider ID from the current origin
+        if (window.isProviderDomain) {
+            return this.get('store')
+                .query('preprint-provider', {
+                    filter: {
+                        domain: `${window.location.origin}/`,
```
Comment (jamescdavis) [inline app/router.js]: API says: Value 'domain' is not a filterable field.

**Comment by brianjgeiger** [comment]
Merge conflicts

**Comment by brianjgeiger** [comment]
@jamescdavis Is this approved with the domain filtering now being in?

**Comment by jamescdavis** [comment]
Looking at it now to make sure.

**Comment by brianjgeiger** [comment]
@binoculars Could you give me a new PR for develop, please?

**Comment by binoculars** [comment]
Sure thing

**Comment by jamescdavis** [comment]
I'd be happy to look at it again against develop. Just ping me.

**Pull request merged**